src/Sarif.Multitool/Rules/SARIF1012.MessageArgumentsMustBeConsistentWithRule.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.CodeAnalysis.Sarif.Multitool.Rules
+{
+    public class MessageArgumentsMustBeConsistentWithRule : SarifValidationSkimmerBase
+    {
+        /// <summary>
+        /// SARIF1012
+        /// </summary>
+        public override string Id => RuleId.MessageArgumentsMustBeConsistentWithRule;
+
+        /// <summary>
+        /// Placeholder
+        /// </summary>
+        public override MultiformatMessageString FullDescription => new MultiformatMessageString { Text = RuleResources.SARIF1012_MessageArgumentsMustBeConsistentWithRule_FullDescription_Text };
+
+        protected override IEnumerable<string> MessageResourceNames => new string[] {
+            nameof(RuleResources.SARIF1012_MessageArgumentsMustBeConsistentWithRule_Error_MessageIdMustExist_Text),
+            nameof(RuleResources.SARIF1012_MessageArgumentsMustBeConsistentWithRule_Error_SupplyEnoughMessageArguments_Text)
+        };
+
+        public override FailureLevel DefaultLevel => FailureLevel.Error;
+
+        private IList<ReportingDescriptor> currentRules;
+
+        protected override void Analyze(Run run, string runPointer)
+        {
+            this.currentRules = run?.Tool?.Driver?.Rules;
+        }
+
+        protected override void Analyze(Result result, string resultPointer)
+        {
+            // if message.id is not null, we have to check if that exists in rules
+            if (!string.IsNullOrEmpty(result.Message.Id))
+            {
+                ReportingDescriptor rule = result.RuleIndex >= 0
+                    ? (this.currentRules?[result.RuleIndex])
+                    : (this.currentRules?.FirstOrDefault(r => r.Id == (result.RuleId ?? result.Rule?.Id)));
+
+                if (this.currentRules == null
+                    || rule == null
+                    || !rule.MessageStrings.ContainsKey(result.Message.Id))
+                {
+                    // {0}: Placeholder {1} {2}
+                    LogResult(
+                        resultPointer,
+                        nameof(RuleResources.SARIF1012_MessageArgumentsMustBeConsistentWithRule_Error_MessageIdMustExist_Text),
+                        result.Message.Id,
+                        (result.RuleIndex >= 0
+                        ? result.RuleIndex.ToString()
+                        : (result.RuleId ?? result.Rule?.Id ?? "null")));
```

Comment:
Here you want to display the rule id, and the SARIF SDK has a helper for that too: `result.ResolveRuleId(run)`. It might return `null`, so you _do_ have to keep the `null` check logic.

Reply:
will push soon

---
In reply to: [447337251](https://github.com/microsoft/sarif-sdk/pull/1944#discussion_r447337251) [](ancestors = 447337251)